tests/pytests/integration/utils/test_smb.py

```diff
@@ -0,0 +1,328 @@
+                "realm = saltstack.com\n"
+                "interfaces = lo 127.0.0.0/8\n"
+                "smb ports = 1445\n"
+                "log level = 2\n"
+                "map to guest = Bad User\n"
+                "enable core files = no\n"
+                "passdb backend = smbpasswd\n"
+                "smb passwd file = {passwdb}\n"
+                "lock directory = {samba_dir}\n"
+                "state directory = {samba_dir}\n"
+                "cache directory = {samba_dir}\n"
+                "pid directory = {samba_dir}\n"
+                "private dir = {samba_dir}\n"
+                "ncalrpc dir = {samba_dir}\n"
+                "socket options = IPTOS_LOWDELAY TCP_NODELAY\n"
+                "min receivefile size = 0\n"
+                "write cache size = 0\n"
+                "client ntlmv2 auth = no\n"
+                "client min protocol = SMB3_11\n"
+                "client plaintext auth = no\n"
+                "\n"
+                "[public]\n"
+                "path = {public_dir}\n"
+                "read only = no\n"
+                "guest ok = no\n"
+                "writeable = yes\n"
+                "force user = {username}\n"
+            )
+        _smbd = subprocess.Popen([shutil.which("smbd"), "-F", "-P0", "-s", samba_conf])
+        time.sleep(1)
+
+        conn_dict = {
+            "tmpdir": tmpdir,
+            "samba_dir": samba_dir,
+            "public_dir": public_dir,
+            "passwdb": passwdb,
+            "username": username,
+            "samba_conf": samba_conf,
+        }
+
+        pidfile = Path(str(samba_dir) + os.sep + "smbd.pid")
+        assert pidfile.exists()
+        with salt.utils.files.fopen(pidfile, "r") as fp:
+            _pid = int(fp.read().strip())
+        if not check_pid(_pid):
+            raise Exception("Unable to locate smbd's pid file")
+
+        yield conn_dict
+
+        log.warning("teardown")
+        os.kill(_pid, signal.SIGTERM)
+
+
+def test_write_file_ipv4(smb_dict):
+    """
+    Transfer a file over SMB
+    """
+    name = "test_write_file_v4.txt"
+    content = "write test file content ipv4"
+    share_path = Path(str(smb_dict["public_dir"]) + os.sep + name)
+    assert not share_path.exists()
+
+    local_path = tempfile.mktemp()
+    with salt.utils.files.fopen(local_path, "w") as fp:
+        fp.write(content)
+    conn = salt.utils.smb.get_conn("127.0.0.1", smb_dict["username"], "foo", port=1445)
+    salt.utils.smb.put_file(local_path, name, "public", conn=conn)
+    conn.close()
+
+    assert share_path.exists()
+    with salt.utils.files.fopen(share_path, "r") as fp:
+        result = fp.read()
+    assert result == content
+
+
+@pytest.mark.skipif(not IPV6_ENABLED, reason="IPv6 not enabled")
+def test_write_file_ipv6(smb_dict):
+    """
+    Transfer a file over SMB
+    """
+    name = "test_write_file_v6.txt"
+    content = "write test file content ipv6"
+    share_path = Path(str(smb_dict["public_dir"]) + os.sep + name)
```

Comment:
```suggestion
    share_path = Path(str(smb_dict["public_dir"]) + os.sep + name)
```
Why are we doing this to paths?!

See all of my fixes for this seen above...

Reply:
doing it all, had the os.sep since originally thinking Windows but :facepalm: , Windows doesn't need smb :laughing: